hazelcast/include/hazelcast/client/serialization/pimpl/SerializationService.h

```diff
@@ -106,12 +140,15 @@ namespace hazelcast {
                             return std::auto_ptr<T>();
                         }
 
-                        // Constant 4 is Data::TYPE_OFFSET. Windows DLL export does not
+                        int32_t typeId = data.getType();
+                        
+                        // Constant 8 is Data::DATA_OFFSET. Windows DLL export does not
                         // let usage of static member.
-                        DataInput dataInput(data.toByteArray(), 4);
+                        DataInput dataInput(data.toByteArray(), 8);
 
```

Comment:
As far as I understand SerializationConstants.checkClassType is not used anymore. 
Can we add that that check here ?